lib/timeline/component/Group.js

```diff
@@ -206,158 +206,174 @@ Group.prototype.getLabelWidth = function() {
-      resized = util.updateProperty(this, 'height', height) || resized;
-      // recalculate size of label
-      resized = util.updateProperty(this.props.label, 'width', this.dom.inner.clientWidth) || resized;
-      resized = util.updateProperty(this.props.label, 'height', this.dom.inner.clientHeight) || resized;
-    }).bind(this));
-
-    queue.push((function () {
-      // apply new height
-      this.dom.background.style.height = height + 'px';
-      this.dom.foreground.style.height = height + 'px';
-      this.dom.label.style.height = height + 'px';
-    }).bind(this));
-
-    queue.push((function () {
-      // update vertical position of items after they are re-stacked and the height of the group is calculated
-      for (var i = 0, ii = this.visibleItems.length; i < ii; i++) {
-        var item = this.visibleItems[i];
-        item.repositionY(margin);
-        if (!this.isVisible && this.groupId != "__background__") {
-          if (item.displayed) item.hide();
-        }
-      }
-
-      if (!this.isVisible && this.height) {
-        return resized = false;
-      }
-
-      return resized;
-    }).bind(this));
+  var resized = false;
+  var lastIsVisible = this.isVisible;
+  var height;
 
-    if (returnQueue) {
-      return queue;
-    } else {
-      var result;
-      queue.forEach(function (fn) {
-        result = fn();
-      });
-      return result;
-    }
-  };
+  var queue = [
+    // force recalculation of the height of the items when the marker height changed
+    // (due to the Timeline being attached to the DOM or changed from display:none to visible)
+    (function () {
+      forceRestack = this._didMarkerHeightChange.bind(this);
+    }).bind(this),
+
+    // recalculate the height of the subgroups
+    this._updateSubGroupHeights.bind(this, margin),
+
+    // calculate actual size and position
+    this._calculateGroupSizeAndPosition.bind(this),
+
+    // check if group is visible
+    (function() {
+      this.isVisible = this._isGroupVisible.bind(this)(range, margin);
+    }).bind(this),
+
+    // redraw Items if needed
+    (function() {
+      this._redrawItems.bind(this)(forceRestack, lastIsVisible, margin, range)
+    }).bind(this),
+
+    // update subgroups
+    this._updateSubgroupsSizes.bind(this),
+
+    // recalculate the height of the group
+    (function() {
+      height = this._calculateHeight.bind(this)(margin);
+    }).bind(this),
+
+    // calculate actual size and position again
+    this._calculateGroupSizeAndPosition.bind(this),
+
+    // check if resized
+    (function() {
+      resized = this._didResize.bind(this)(resized, height)
+    }).bind(this),
+
+    // apply group height
+    (function() {
+      this._applyGroupHeight.bind(this)(height)
+    }).bind(this),
+
+    // update vertical position of items after they are re-stacked and the height of the group is calculated
+    (function() {
+      return resized = this._updateItemsVerticalPosition.bind(this)(resized, margin)
```

Comment:
For my understanding: this is the last function in the queue and it returns something. The return value is collected [here](https://github.com/yotamberk/vis/blob/329d9817f94d53738222be9ae361633e3a912209/lib/timeline/component/ItemSet.js#L701). Is this the intention?

Reply:
I suppose the assignment to `resize` isn't really necessary for the return, but it's kept in for completeness. I guess this is OK (I would probably do the same).